guida/src/app/components/PostForm/PostForm.tsx

```diff
@@ -51,6 +59,22 @@ const PostForm: React.FC<PostFormProps> = ({ post, submitText, onSuccess, onClos
                 />
                 {titleError && <p className={styles.error}>{titleError}</p>}
               </div>
+              <div className={styles.postFormElement}>
+                <label>
+                    Publish as Role
+                </label>
+                <select
+                  value={Number(selectedAuthorRole)}
+                  onChange={(e) => {setSelectedAuthorRole(Number(e.target.value));}}
+                >
+                <option value={undefined}>None</option>
+                {userRoles.map((role) => (
+                  <option key={role.label} value={role.value}>
+                    {role.label}
+                  </option>
+                ))}
+                </select>
+              </div>
```

Comment:
No Css? 👀

Reply:
Tried to fix this but couldn't see how without refactoring to a seperate component and I don't want to. Looks ok as is.